src/extensions/extension-discovery/is-compatible-extension/is-compatible-extension.ts

```diff
@@ -9,44 +9,14 @@ interface Dependencies {
   appSemVer: SemVer;
 }
 
-export const isCompatibleExtension = ({
-  appSemVer,
-}: Dependencies): ((manifest: LensExtensionManifest) => boolean) => {
-  const { major, minor, patch, prerelease: oldPrelease } = appSemVer;
-  let prerelease = "";
-
-  if (oldPrelease.length > 0) {
-    const [first] = oldPrelease;
-
-    if (first === "alpha" || first === "beta" || first === "rc") {
-      /**
-       * Strip the build IDs and "latest" prerelease tag as that is not really
-       * a part of API version
-       */
-      prerelease = `-${oldPrelease.slice(0, 2).join(".")}`;
-    }
-  }
-
-  /**
-   * We unfortunately have to format as string because the constructor only
-   * takes an instance or a string.
-   */
-  const strippedVersion = new SemVer(
-    `${major}.${minor}.${patch}${prerelease}`,
-    { includePrerelease: true },
-  );
+export const isCompatibleExtension = (deps: Dependencies): ((manifest: LensExtensionManifest) => boolean) => {
+  const { major: appMajor, minor: appMinor } = deps.appSemVer;
 
   return (manifest: LensExtensionManifest): boolean => {
-    if (manifest.engines?.lens) {
-      /**
-       * include Lens's prerelease tag in the matching so the extension's
-       * compatibility is not limited by it
-       */
-      return semver.satisfies(strippedVersion, manifest.engines.lens, {
-        includePrerelease: true,
-      });
-    }
+    const { major: extMajor, minor: extMinor } = new SemVer(manifest.engines.lens, {
```

Comment:
It would be really good to have some tests for this code nonetheless. For instance what happens if an extension has the string `>=5.5.0`? I think this code might throw in that case.

Reply:
True. Already noticed versions starting with `^` throws in semver constructor, so something need to be done before passing raw version from extension/app (normalize version to not throwable or maybe `Semver` has some built-in options).
Anyway I'll try to add some unit tests to check that different ext/app version range formats wont break this function.